Check macOS builds

I've been fiddling with the build system trying to resolve issues on BSD / NixOS. Basically I've been cleaning up the autoconf hacks and using more orthodox methods trying to root out why the tarballs sometimes can't be built without running `autoreconf` (which shouldn't be necessary for distribution sources, only to bootstrap the Git sources).

In the process it's possible that I broke Darwin / macOS builds. Unfortunately my ability to test that platform is pretty chaotic. I have a VirtualBox machine spun up that *mostly* works. It was Catalina, now updated to Big Sur. Unfortunately I don't really understand my way around the platform and have just been jabbing stuff at it. I can't currently build SILE, but I don't know if the culprit is a broken dev environment or if the SILE sources are un-buildable.

Trying `brew install sile --HEAD` fails with this:

![image](https://user-images.githubusercontent.com/173595/108620240-e17ead00-743b-11eb-9744-4140663577d9.png)

Am I missing some headers / libraries? Can you build the current `master` right now @simoncozens?

This is the nub of it:

```
  OBJC     macfonts_la-macfonts.lo
While building module 'AppKit' imported from macfonts.m:1:
...
In file included from <module-includes>:76:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/IOKit.framework/Headers/scsi/SCSICmds_INQUIRY_Definitions.h:92:9: error:
      expected member name or ';' after declaration specifiers
        UInt8           VERSION...
        ~~~~~           ^
<command line>:8:17: note: expanded from here
#define VERSION "0.10.14.r67-g3a10d56-dirty"
```

A simple "#undef version" on line 1 of macfonts.m does not fix it. I don't know how the VERSION macro is getting into the DEFS in the automake-generated `src/Makefile`.

--- a/Makefile-fonts
+++ b/Makefile-fonts
@@ -115,7 +115,7 @@ TESTFONTFILES += SBL_Hbrw.ttf
 	touch $@
 
 .fonts/DejaVu%: .sources/dejavu-fonts-ttf-2.37.tar.bz2 | .fonts
-	bsdtar -x -f $< -C $(dir $@) --strip-components 2 $(basename $(<F))/ttf/$(notdir $@)
+	bsdtar -x -f $< -C $(dir $@) --strip-components 2 $(basename $(basename $(<F)))/ttf/$(notdir $@)
 	touch $@
 
 .fonts/GenB%: .sources/GentiumBasic_1102.zip | .fonts

--- a/Makefile-luarocks
+++ b/Makefile-luarocks
@@ -4,8 +4,9 @@ if !SYSTEM_LUAROCKS
 LUAMODLOCK := sile-dev-1.rockslock
 
 LUAROCKS := luarocks --tree lua_modules --lua-version $(LUA_VERSION)
+TMPFILE != mktemp
 genrockslock := $(LUAROCKS) $(LUAROCKSARGS) list --porcelain | awk '{print $$1 " " $$2}'
-rocksmatch := cmp -s $(LUAMODLOCK) <($(genrockslock))
+rocksmatch := ( T=$$(mktemp); trap 'rm -f "$$T"' EXIT HUP TERM; $(genrockslock) > "$$T"; cmp -s $(LUAMODLOCK) "$$T" )
 
 installrocks: $(LUAMODLOCK) $(shell $(rocksmatch) || echo lua_modules)
 

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -31,7 +31,7 @@ if APPKIT
 pkglib_LTLIBRARIES += macfonts.la
 macfonts_la_SOURCES = macfonts.m
 macfonts_la_LDFLAGS = -module -avoid-version -shared $(UNDEFINED)
-macfonts_la_OBJCFLAGS = $(FREETYPE_CFLAGS) $(LUA_INCLUDE)  -fmodules
+macfonts_la_OBJCFLAGS = -U VERSION $(FREETYPE_CFLAGS) $(LUA_INCLUDE)  -fmodules
 macfonts_la_LIBADD = $(FREETYPE_LIBS) $(MY_LUA_LIB)
 to_copy += .libs/macfonts.@SHARED_LIB_EXT@
 endif

